app/controllers/merchant/items_controller.rb

```diff
@@ -46,6 +46,17 @@ def activate
     redirect_to '/merchant/items'
   end
 
+  def destroy
+    item = Item.find(params[:id])
+    if item.orders.empty?
+      item.destroy
+      flash[:success] = "Yer item walked the plank fer good!"
+    else
+      flash[:failure] = "ARGH! Ye shan't be deletin booty that's already been plundered!"
```

Comment:
A) this is incredible
B) we should probably include a little bit in plainspeak about what the error means, too. :D